### PR TITLE
Support multi-tenancy: load organization-specific configuration from file

### PR DESCRIPTION
**Pull request by anaulin**
This PR addresses https://github.com/starlinglab/starling-integrity-api/issues/54. In particular:
* loads a JSON file with configuration for different organizations, keyed by a unique `organization_id`
* `FsWatcher` is now organization specific, and uses the `organization_id` as a prefix in all its directories
  * this means that we now run one process (in the "Python multiprocess" sense) for each organization; if we end up with dozes of organizations, we will probably want to split out the file watchers into a separate service, or even run each organization file watcher separately
* claim generation uses organization-specific creative work author information loaded from file

This PR includes an example configuration JSON file, and an updated JWT example with an organization id.

One important to note assumption made here is that the `organization_id` should be filename-compatible, as it will be used as a directory name prefix. The code will try to check for this and reject organization ids that are not filename-compatible at startup, to try to catch any mistakes early.

### CODE REVIEW

File: jwt-payload.json.example [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+    "organization_id": "hyphacoop",
```
Comment (anaulin) [inline jwt-payload.json.example:2]: @benhylau FYI this is my proposed extension to our JWT format so far. Let me know what you think.

**Review by benhylau — APPROVED** [review]

File: jwt-payload.json.example (hunk 1 above)
Comment (benhylau) [reply jwt-payload.json.example:2]: Just need to know if we can use `-`. Either way I think it's okay, but having `-` may be easier to read.

Comment (anaulin) [reply jwt-payload.json.example:2]: Yes, we can use `-` or `_`. In general, anything that you'd put in a directory or file name.

File: starlingcaptureapi/asset_helper.py [hunk 2]
```diff
@@ -1,222 +1,176 @@
...
+    def __init__(self, organization_id):
+        """
+        Args:
+            organization_id: string uniquely representing an organization
+                must not contain spaces or special characters, as it will become
+                part of directory names (e.g. "hyphacoop" good, not "Hypha Coop")
+        """
+        if self._filename_safe(organization_id) != organization_id:
+            raise ValueError(
+                f"Organization {organization_id} does not to be filename safe!"
```
Comment (YurkoWasHere) [inline starlingcaptureapi/asset_helper.py:21]:
```suggestion
                f"Organization {organization_id} does not have to be filename safe!"
```
Not sure of logic if `does not have` or `does have`

Comment (anaulin) [reply starlingcaptureapi/asset_helper.py]: Oops, yes, I fixed the error. It was meant to say "it is not filename safe".

**Comment by benhylau** [comment]
We have a field capture project on Monday, and since `CREATIVE_WORK_AUTHOR` is not used in the capture claim, we are okay keeping the existing code for the session. It just means a bit more manual migration to untangle the two projects' photos in same directory.

@YurkoWasHere let's plan to deploy this Tuesday so it can be ready for a bunch of capture sessions from Thursday onwards, alongside the Dropbox sync.

@anaulin tks for this. I will have this reviewed and tested locally by eod Monday.

File: tests/test_claim.py [hunk 3]
```diff
@@ -217,12 +213,26 @@ def test_generates_create_claim_with_partial_reverse_geocode(reverse_geocode_moc
...
+                    "@id": "https://twitter.com/somefakestuff",
+                    "@type": "Organization",
+                    "identifier": "https://example.com",
+                    "name": "hyphacoop",
+                }
+            ]
+        }
+    }
+
+    claim = _claim.generate_store("a-made-up-cid", "hyphacoop")
```
Comment (benhylau) [inline tests/test_claim.py:235]: Think these have to match bc of how Adobe's verifier renders the text.

```suggestion
                    "@id": "https://twitter.com/example",
                    "@type": "Organization",
                    "identifier": "https://example.com",
                    "name": "example",
                }
            ]
        }
    }

    claim = _claim.generate_store("a-made-up-cid", "example")
```

File: tests/test_claim.py [hunk 4]
```diff
@@ -217,12 +213,26 @@ def test_generates_create_claim_with_partial_reverse_geocode(reverse_geocode_moc
 
 
 def test_generates_update_claim():
-    claim = _claim.generate_update()
+    claim = _claim.generate_update("hyphacoop")
```
Comment (benhylau) [inline tests/test_claim.py:216]:
```suggestion
    claim = _claim.generate_update("example")
```

File: tests/test_asset_helper.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+
+def test_rejects_non_file_safe_org():
+    with pytest.raises(ValueError, match="filename safe"):
+        asset_helper.AssetHelper("not File Safe")
+
+
+def test_directory_contains_org():
+    helper = asset_helper.AssetHelper("hyphacoop")
+    assert helper.get_claims_internal() == "/tests/assets_dir/hyphacoop/claims"
```
Comment (benhylau) [inline tests/test_asset_helper.py:13]:
```suggestion
    helper = asset_helper.AssetHelper("example")
    assert helper.get_claims_internal() == "/tests/assets_dir/example/claims"
```

File: config.example.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            "@type": "Organization",
+            "credential": [],
+            "identifier": "https://hypha.coop",
+            "name": "Hypha Coop"
+        },
+        {
+            "@id": "https://twitter.com/hyphacoop",
+            "@type": "Organization",
+            "identifier": "https://hypha.coop",
+            "name": "hyphacoop"
```
Comment (benhylau) [inline config.example.json:16]: Matching JWT example.

```suggestion
            "name": "HyphaCoop"
```

**Comment by benhylau** [comment]
@anaulin this looks great! Before reading this, I was wondering how many classes will have to turn into per org instances :)

There is a weird thing with naming Twitter blocks that aren't "spec" but the way things are rendered, the text have to match up. I changed up some tests and examples. Looks like I broke the test too. I'll fix, merge, and deploy later today. Thanks!

**Pull request merged**